routes/posts.js

```diff
@@ -1,15 +1,21 @@
-const express = require("express")
+const express = require("express");
 const router = express.Router();
-const data = require("../data/index")
-const movies = data.movies;
+const data = require("../data/index");
+const moviesData = data.movies;
 const postsData = data.posts;
 
-
 // This should render a page to create a post
 router.get("/movieSelection", async (req, res) => {
-    res.render("partials/createPost", {title: "Create a post"})
-})
-router.post("/", async (req, res) => {
-    
-})
-module.exports = router
\ No newline at end of file
+  res.render("partials/createPost", { title: "Create a post" });
+});
+router.get("/:id", async (req, res) => {
+  const post = await postsData.getPost(req.params.id);
+  const movie = await moviesData.getMovie(post.postMovieId);
+  res.render("partials/postDetail", {
+    title: post.postTitle,
+    post: post,
+    movie: movie,
+  });
+});
+router.post("/", async (req, res) => {});
+module.exports = router;
```

Comment:
router.get("/:id", async (req, res) => {
  try {
    let postList = await postsData.getPost(req.params.id);
    let allComments = commentsData.getAllComments(req.params.id);
    res
      .status(200)
      .render("partials/postPage", { post: postList, comments: allComments });
  } catch (e) {
    res
      .status(404)
      .redirect("partials/allPosts", { errorMessage: "Post could be found" });
  }
});

This is the fucntion i wrote for get post by id, can you try and use this? cause i will need comments of the post as well. or try and integrate maybe?